guide/adopting-the-configuration-cache.md

```diff
@@ -0,0 +1,384 @@
+|         |    |    `--- :processResources                (org.gradle.language.jvm.tasks.ProcessResources)
+|         |    `--- :compileJava                          (org.gradle.api.tasks.compile.JavaCompile)
+|         `--- :processTestResources                      (org.gradle.language.jvm.tasks.ProcessResources)
+...
+```
+
+#### Task graph of `./gradlew assemble`
+```
+:assemble                        (org.gradle.api.DefaultTask)
+`--- :jar                        (org.gradle.api.tasks.bundling.Jar)
+     +--- :classes               (org.gradle.api.DefaultTask)
+     |    +--- :compileJava      (org.gradle.api.tasks.compile.JavaCompile)
+     |    `--- :processResources (org.gradle.language.jvm.tasks.ProcessResources)
+     `--- :compileJava           (org.gradle.api.tasks.compile.JavaCompile)
+```
+
+#### Task graph of `./gradlew build`
+```
+:build                                                         (org.gradle.api.DefaultTask)
++--- :assemble                                                 (org.gradle.api.DefaultTask)
+|    `--- :jar                                                 (org.gradle.api.tasks.bundling.Jar)
+|         +--- :classes                                        (org.gradle.api.DefaultTask)
+|         |    +--- :compileJava                               (org.gradle.api.tasks.compile.JavaCompile)
+|         |    `--- :processResources                          (org.gradle.language.jvm.tasks.ProcessResources)
+|         `--- :compileJava                                    (org.gradle.api.tasks.compile.JavaCompile)
+`--- :check                                                    (org.gradle.api.DefaultTask)
+   +--- :checkstyleMain                                      (org.gradle.api.plugins.quality.Checkstyle)
+   |    +--- ...
+   +--- :checkstyleTest                                      (org.gradle.api.plugins.quality.Checkstyle)
+   |    +--- ...
+   +--- :spotlessCheck                                       (org.gradle.api.DefaultTask)
+   |    `--- ...
+   +--- :test                                                (org.gradle.api.tasks.testing.Test)
+   |    `--- ...
+   ...
+```
+
+When you first run a gradle task, the task graph is serialized and stored on disk. Upon a subsequent run of the same task, If none of these inputs have changed, Gradle skips the configuration phase entirely, loads the task graph from disk, and goes straight to the execution phase.
+
+The configuration phase typically runs faster than execution since heavy work belongs in the latter. However, without the cache, configuration phase is rerun with every single Gradle run. If you run a unit test multiple times, configuration is repeated, reproducing the same task graph every time. When the task itself is light, configuration can take up a large fraction of the latency. Configuration caching solves this by storing and reusing configuration results between runs, eliminating redundant work and speeding up iteration cycles.
```

Comment:
Feel like you're underselling it a bit here as well. Why is unit testing *such* a good example? because it requires a **tight feedback loop**. The task can be light, or medium, but it can still feel like it takes a while before anything "happens" i.e. paying for configuration.

Reply:
This comment clarifies what you mean! Added.